src/mtev_conf.h

```diff
@@ -44,6 +44,26 @@
 
 typedef void * mtev_conf_section_t;
 
+typedef enum {
+  MTEV_PARAM_INT8,
+  MTEV_PARAM_UINT8,
+  MTEV_PARAM_INT16,
+  MTEV_PARAM_UINT16,
+  MTEV_PARAM_INT32,
+  MTEV_PARAM_UINT32,
+  MTEV_PARAM_INT64,
+  MTEV_PARAM_UINT64,
+  MTEV_PARAM_FLOAT,
+  MTEV_PARAM_DOUBLE,
+  MTEV_PARAM_BOOLEAN
+} mtev_param_type_t;
+
+typedef mtev_boolean (*mtev_param_validator_t)(mtev_param_type_t, void *mem);
+mtev_boolean
+mtev_conf_register_global_param(const char *name, const char *xpath,
+                                mtev_param_type_t ptype, void *mem,
+                                mtev_param_validator_t vf);
+
```

Comment:
I'd like to toss the confstr stuff.  It's used inconsistently and didn't take into consideration the complexities that the current config system handles.

The idea of a parser sounds fairly useful, but a parse has to look at text which makes the common case of simply validating a parsed value much harder and more repetitive on the caller.  So if we add a parser, it should be in addition to the validator -- which only validates the actual value.